src/arith/const_int_bound.cc

```diff
@@ -150,10 +150,12 @@ class ConstIntBoundAnalyzer::Impl :
       const PrimExprNode* op = expr.as<PrimExprNode>();
       auto val = bound_->find(op);
       if (val != bound_->end()) {
-        CHECK(val->second->min_value == res.min_value &&
-              val->second->max_value == res.max_value)
-          << "Detected bound for " << expr
-          << "conflicts with memorization";
+        auto everything = Everything(op->dtype);
+        CHECK(
+            (val->second->min_value == res.min_value && val->second->max_value == res.max_value) ||
+            (val->second->min_value == everything.min_value &&
```

Comment:
I guess this is for the case where `override == true`, right? If so, what about make it explicit and just checks if override is true? Or can we clear the whole memo map when updating with override?

Reply:
Good catch.  I forgot to bring this up for discussion.

Reply:
It is not exactly override, but assuming a partial order that limit bound is better than inf bound.  I still keep assertion for different limit bounds, when override == false.
This change alters API semantics, but I am not sure if this behavior worth another mode in addition to override.

Reply:
I agree that override with partial order is good. Can we check if `res` is contained in `val->second` ? (`val->second` being everything is a special case of this)

Reply:
@hzfan do you mean to update val->second, when res is a subset?  It is a looser check than the current change.  I haven't seen any case that a limit bound improves from a looser limit bound.  If we don't have any such case now, I'd like to let this check assert for people to justify when they change the behavior.